### PR TITLE
3073 message delivery statuses

### PR DESCRIPTION
**Pull request by estellecomment**
:yay: 
medic/medic-webapp#3073

### CODE REVIEW

File: lib/views.js [hunk 1]
```diff
@@ -438,7 +437,7 @@ exports.tasks_messages = {
             // used for fetching a specific message based on uuid
             emit(msg.uuid, val);
```
Comment (garethbowen) [inline lib/views.js]: Can you do an audit and check if we're using this any more?

File: lib/views.js [hunk 2]
```diff
@@ -438,7 +437,7 @@ exports.tasks_messages = {
...
             // used for querying latest tasks in a specific state
-            emit([task.state, when], val);
+            emit(task.state, val);
```
Comment (garethbowen) [inline lib/views.js]: I think it's useful to keep the `when` value - it sorts the results so we can send the oldest first.

Comment (garethbowen) [inline lib/views.js]: NB - the [API documentation](https://github.com/medic/medic-api#get-apiv1messages) says:

> Returns list of messages, oldest first based on timestamp or due date.

I don't think this is true any more.

File: lib/views.js (hunk 1 above)
Comment (estellecomment) [inline lib/views.js]: yeah we are. 
 - Api end point for getting single message by id: `GET /api/v1/messages/:id`
 - Also `updateMessage` uses it:
    - api uses it internally in sms protocol to update state (`api/controllers/sms-gateway.js`, l.47). 
    -  externally `PUT /api/v1/messages/state/:id'`

**Pull request merged**